contracts/vaults/RibbonThetaVault.sol

```diff
@@ -434,6 +435,40 @@ contract RibbonThetaVault is OptionsVaultStorage {
         transferAsset(msg.sender, amount);
     }
 
+    /**
+     * @notice Initiates a withdrawal that can be processed once the round completes
+     * @param shares is the number of shares to withdraw
+     */
+    function initiateWithdraw(uint128 shares) external nonReentrant {
+        require(shares > 0, "!shares");
+
+        // This caches the `round` variable used in shareBalances
+        uint16 currentRound = round;
+        Vault.Withdrawal memory withdrawal = withdrawals[msg.sender];
+
+        require(!withdrawal.initiated, "Existing withdraw");
+
+        (uint256 heldByAccount, uint256 heldByVault) =
+            shareBalances(msg.sender);
+
+        uint256 totalShares = heldByAccount.add(heldByVault);
+
+        require(shares <= totalShares, "Insufficient balance");
+
+        emit InitiateWithdraw(msg.sender, shares, currentRound);
+
+        withdrawals[msg.sender].initiated = true;
+        withdrawals[msg.sender].round = currentRound;
+        withdrawals[msg.sender].shares = shares;
+
+        queuedWithdrawShares = queuedWithdrawShares.add(shares);
+
+        if (shares > heldByVault) {
```

Comment:
shouldn't it be shares > heldByAccount? if the shares is greater than held by user then we will need to transfer shares from vault to user.